pallets/automation-time/src/lib.rs

```diff
@@ -955,6 +1090,80 @@ pub mod pallet {
 				.saturating_add(<T as Config>::WeightInfo::run_xcmp_task())
 		}
 
+		/// Executes auto compounding delegation and reschedules task on success
+		pub fn run_auto_compound_delegated_stake_task(
+			delegator: T::AccountId,
+			collator: T::AccountId,
+			account_minimum: BalanceOf<T>,
+			frequency: Seconds,
+			task_id: T::Hash,
+			mut task: Task<T>,
+		) -> (Task<T>, Weight) {
+			match Self::compound_delegator_stake(
+				delegator.clone(),
+				collator.clone(),
+				account_minimum,
+				Self::calculate_execution_fee(&task.action, 1),
+			) {
+				Ok(delegation) =>
+					Self::deposit_event(Event::SuccesfullyAutoCompoundedDelegatorStake {
+						task_id,
+						amount: delegation,
+					}),
+				Err(e) => {
+					Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+						task_id,
+						error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+						error: e,
+					});
+					return (
+						task,
+						<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+					)
+				},
+			}
+
+			let new_execution_times: Vec<UnixTime> =
+				task.execution_times.iter().map(|when| when.saturating_add(frequency)).collect();
+			let _ = Self::reschedule_existing_task(
+				task_id,
+				task.owner_id.clone(),
+				&task.action,
+				new_execution_times.clone(),
+			)
+			.map(|_| {
+				let new_executions_left: u32 = new_execution_times.len().try_into().unwrap();
+				task.executions_left += new_executions_left;
+				new_execution_times.iter().try_for_each(|t| task.execution_times.try_push(*t))
+			})
+			.map_err(|e| {
+				let err: DispatchErrorWithPostInfo = e.into();
+				Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+					task_id,
+					error_message: Into::<&str>::into(err).as_bytes().to_vec(),
+					error: err,
+				});
+			});
+
+			(task, <T as Config>::WeightInfo::run_auto_compound_delegated_stake_task())
+		}
+
+		fn compound_delegator_stake(
+			delegator: T::AccountId,
+			collator: T::AccountId,
+			account_minimum: BalanceOf<T>,
+			execution_fee: Result<BalanceOf<T>, DispatchError>,
+		) -> Result<BalanceOf<T>, DispatchErrorWithPostInfo> {
+			let reserved_funds = account_minimum.saturating_add(execution_fee?);
```

Comment:
they already paid for this execution, so why would we not up their bond then not schedule the next?

Reply:
See- https://github.com/OAK-Foundation/OAK-blockchain/pull/180#discussion_r923901660

This method will never throw an error.

Reply:
My mistake on what you meant, added todo for this edge- https://github.com/OAK-Foundation/OAK-blockchain/pull/180/commits/8c00dc65f1a0624959ff976cf951be6198ef010f